Library/OcBootManagementLib/BootEntryManagement.c

```diff
@@ -288,19 +294,137 @@ ReadEntryVisibility (
     return BootEntryNormal;
   }
 
-  if (AsciiStrnCmp (Visibility, "Disabled", L_STR_LEN ("Disabled")) == 0) {
-    FreePool (Visibility);
-    return BootEntryDisabled;
+  //
+  // Set some base initialisation.
+  // 'QualifierString' not strictly required as use is guarded by
+  // 'CheckQualifiers' but this appears easy to be missed out in future.
+  //
+  QualifierString = NULL;
+  CheckQualifiers = FALSE;
+
+  if (AsciiStrStr (Visibility, "@") != NULL) {
+    //
+    // contentVisibility qualifier boundary character found.
+    // Get OC instance perent folder name for search string.
+    //
+    OCFolderName = NULL;
+    LoadedImage  = NULL;
+    Status       = gBS->HandleProtocol (
+                          gImageHandle,
+                          &gEfiLoadedImageProtocolGuid,
+                          (VOID **)&LoadedImage
+                          );
+    if (!EFI_ERROR (Status)) {
+      AbsPath = AbsoluteDevicePath (LoadedImage->DeviceHandle, LoadedImage->FilePath);
+      if (AbsPath != NULL) {
+        OCFolderName = OcCopyDevicePathFullName (AbsPath, NULL);
+        FreePool (AbsPath);
+      }
+    }
+
+    if (OCFolderName != NULL) {
+      CheckQualifiers = UnicodeGetParentDirectory (OCFolderName);
+    }
+
+    if (CheckQualifiers == TRUE) {
```

Comment:
For all booleans, we simplify use `if (CheckQualifiers)` and `if (!CheckQualifiers)`

Reply:
Thanks ... updated